crates/sui-sdk/examples/event_api.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+mod utils;
+use crate::utils::example_utils::{setup_for_read, split_coin_digest};
+use futures::stream::StreamExt;
+use sui_sdk::rpc_types::EventFilter;
+use sui_sdk::SuiClientBuilder;
+
+// This example showcases how to use the Event API.
+// At the end of the program it subscribes to the events
+// on the Sui testnet and prints every incoming event to
+// the console. The program will loop until it is force
+// stopped.
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    let (sui, active_address) = setup_for_read().await?;
+
+    println!(" *** Get events *** ");
+    // for demonstration purposes, we set to make a transaction
+    let digest = split_coin_digest(&sui, &active_address).await?;
```

Comment:
Because you are creating a transaction, I think you want to use `setup_for_write` in this example.

Reply:
Sure, we can do that.